compiler/modules/front.syntax/tests/posTree.bal

```diff
@@ -0,0 +1,237 @@
+import wso2/nballerina.comm.diagnostic as d;
+
+type TerminalSyntaxNode TerminalSyntaxAstNode|FixedSyntaxNode|IdentifierSyntaxNode|StringLiteralSyntaxNode;
+type AstSyntaxNode TerminalSyntaxAstNode|NonTerminalSyntaxNode;
+type SyntaxNode NonTerminalSyntaxNode|TerminalSyntaxNode;
+
+type TerminalSyntaxAstNode record {|
+    AstNode astNode;
+    string token;
+|};
+
+type NonTerminalSyntaxNode record {|
+    AstNode astNode;
+    SyntaxNode[] childNodes;
+|};
+
+type RootSyntaxNode record {|
+    ModulePart part;
+    SyntaxNode[] childNodes;
+|};
+
+// represents keywords and fixed tokens
+type FixedSyntaxNode record {|
+    FixedToken token;
+    Position? pos = ();
+|};
+
+type StringLiteralSyntaxNode record {|
+    string literal;
+    Position? pos = ();
+|};
+
+// represents identifiers
+type IdentifierSyntaxNode record {|
+    string name;
+    Position? pos;
+|};
+
+type AstNode record {|
+   *PositionFields;
+   any...;
+|};
+
+function validateModulePart(ModulePart part) {
+    RootSyntaxNode root = buildTree(part);
+    checkpanic validateSyntaxNode(root, new(part.file));
+}
+
+function buildTree(ModulePart part) returns RootSyntaxNode {
```

Comment:
Goes in syntaxNode.bal and called rootSyntaxNode.

Reply:
fixed